internal/provider/example_data_source.go

```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+var _ tfsdk.DataSourceType = exampleDataSourceType{}
+var _ tfsdk.DataSource = exampleDataSource{}
```

Comment:
since some folks maybe new to Go - is it be worth adding a comment as to why these are here?

Reply:
Excellent idea -- thanks!